processing/src/main/java/io/druid/query/QueryInterruptedException.java

```diff
@@ -22,27 +22,87 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.TimeoutException;
+
 public class QueryInterruptedException extends RuntimeException
 {
-  public QueryInterruptedException() {
-    super();
-  }
+  public static final String QUERY_INTERRUPTED = "Query interrupted";
+  public static final String QUERY_TIMEOUT = "Query timeout";
+  public static final String QUERY_CANCELLED = "Query cancelled";
+  public static final String UNKNOWN_EXCEPTION = "Unknown exception";
+
+  private static final List<String> listKnownException = Arrays.asList(
+      QUERY_CANCELLED,
+      QUERY_INTERRUPTED,
+      QUERY_TIMEOUT,
+      UNKNOWN_EXCEPTION
+  );
+
+  @JsonProperty
+  private final String causeMessage;
+  @JsonProperty
+  private final String host;
 
   @JsonCreator
-  public QueryInterruptedException(@JsonProperty("error") String message)
+  public QueryInterruptedException(
+      @JsonProperty("error") String message,
+      @JsonProperty("causeMessage") String causeMessage,
+      @JsonProperty("host") String host
+  )
   {
     super(message);
+    this.causeMessage = causeMessage;
+    this.host = host;
   }
 
   public QueryInterruptedException(Throwable cause)
   {
-    super(cause);
+    this(cause, null);
+  }
+
+  public QueryInterruptedException(Throwable e, String host)
+  {
+    super(e);
+    this.host = host;
+    causeMessage = e.getMessage();
   }
 
   @JsonProperty("error")
   @Override
   public String getMessage()
   {
-    return super.getMessage();
+    if (this.getCause() == null) {
```

Comment:
@drcrallen and @cheddar the current UT test this branch by this [code](https://github.com/b-slim/druid/blob/77925cc06101aea3bf1e7f96173e825d0336ae40/server/src/main/java/io/druid/client/DirectDruidClient.java#L484-L484) if you have another idea in mind please let me know